amount.go

```diff
@@ -120,6 +120,16 @@ func MultiplyByFloat(amount AmountBlockchain, multiplier float64) AmountBlockcha
 	return result
 }
 
+// NewAmountBlockchainFromUint64 creates a new AmountBlockchain from a uint64
+func NewAmountBlockchainFromInt64(i64 int64) AmountBlockchain {
+	abs := uint64(i64)
+	if i64 < 0 {
+		abs = uint64(-i64)
```

Comment:
I think this isn't really behavior we want.  If you make a negative amount, you would expect `AmountBlockchain` to also be negative.

If this isn't the behavior, I would change the function signature to be:

```go
func NewAmountBlockchainFromInt64(i64 int64) (amount AmountBlockchain, ok bool) {
```

If it's negative, return `0, false` -- indicating that it failed and the default value (0) is used.